src/legacy/ui/public/vis/agg_configs.ts

```diff
@@ -209,14 +219,48 @@ class AggConfigs extends IndexedArray {
     return dslTopLvl;
   }
 
-  getRequestAggs() {
-    //collect all the aggregations
-    const aggregations = this.reduce((requestValuesAggs, agg) => {
+  getAll() {
+    return [...this.aggs];
+  }
+
+  getByIndex(index: number) {
+    return this.aggs[index];
+  }
+
+  getById(id: string) {
+    return this.aggs.find(agg => agg.id === id);
+  }
+
+  getByName(name: string) {
+    return this.aggs.filter(agg => agg.type.name === name);
+  }
+
+  byType(type: string) {
+    return this.aggs.filter(agg => agg.type.type === type);
+  }
+
+  bySchemaName(schema: string) {
+    return this.aggs.filter(agg => agg.schema && agg.schema.name === schema);
+  }
+
+  bySchemaGroup(group: string) {
+    return this.aggs.filter(agg => agg.schema && agg.schema.group === group);
+  }
+
```

Comment:
One other thing I'm seeing repeated a lot in this PR is finding params, metrics, and buckets by name. I don't want to add more than is needed to the API here, but I wonder if it would be worth adding methods for those if it will be a common use case? 

e.g. `byMetricName`, `byBucketName`, `byParamName`